### ISSUE TITLE
Should we remove the included GWT SDKs from the plugin?

### ISSUE THREAD

**Issue by keinhaar** [gwt-plugins/gwt-eclipse-plugin#458]
Why we maybe should remove the SDKs.
- I think nobody is using all of the included SDKs at once
- GWT 2.10 is missing
- Only the main Versions are included. GWT 2.8.2 for example is missing.
- Increases the download time of the plugin
- Increases the build time
- Increases the disk consumption
- It's easy to specify your currently used version

Maybe somebody could create an Popup that asks for a GWT version if no SDK is configured.

Any thoughts on this?

**Comment by protoism** [comment]
@keinhaar I don't get the point. Plugin download is opt-in, you're not forced to download an SDK. 
Let's suppose you don't download any SDK with the plugin, nor any stand-alone SDK, what should happen?

* Maven / Gradle care about classpath, and Plugin simply finds the SDK in the classpath
* The plugin exposes a "virtual" container, which downloads the SDK when needed
* Other options I did not think of

**Comment by niloc132** [comment]
On the other hand, many users just check all and install. And the install size is huge on the server, to include a new copy of every old version of GWT on each update. Eventually that will add up.

**Comment by keinhaar** [comment]
@protoism There are already some Warnings about a missing SDK from the Plugin
- On the Console: "GWT SDK not installed."
- In the Error Log: "GWT SDK not found"
- In the Problems View: "The project XXXX does not have any GWT SDKs on it's build path"
- In the Problems View: "Unbound classpath container"...

We could create a compound update site, so the SDK Plugins could be separated. That would still allow the users to install the SDK (also different versions) together with the plugin, but the SDKs would not be inside the gwt plugin itself. 
Each SDK version would be only one time on the server. The same way could be used to reintegrate SDBG with an always current version. I don't know if the Marketplace accepts compound update sites, but i don't see any reason why it should not.
By the way, what is the Update Site URL used for the Marketplace entry? 

Or we could do as @niloc132 said, and add just the latest SDK. But that does not solve the problem completely, because then we would need to update the plugin, if a new SDK version is released.

**Comment by protoism** [comment]
@keinhaar so you don't want to overhaul the plugin architecture,... It's just about SDK distribution.
@niloc132 is pushing a bit more: the "classpath", in a maven/gradle build already contains an SDK, so there is really no need for an eclipse delivered SDK

Here what a maven project produces
![image](https://user-images.githubusercontent.com/1098792/222094330-f3c52b69-adc3-4f1d-8286-df2a59e9de63.png)

I only wonder if the "Use specific SDK" might be replaced by something like "use the SDK which is already in the classpath".

Summarizing:
* The plugin can contribute with a GWT container, which is a classpath contributor. This container is only needed by non maven / gradle projects
* The plugin still needs an SDK, which is more than the bunch of GWT jars already in the classpath, 

My doubt is: what is the SDK (not the GWT container) used for?

**Comment by keinhaar** [comment]
> so you don't want to overhaul the plugin architecture,... It's just about SDK distribution.
Yes. At my company we have the SDK separately downloaded (version 2.10), and don't need the packaged version, but for beginners, it may be easier if they could just start by installing the plugin and the SDK at once. And yes maven does download the SDK anyway, but not every project is maven based.

**Comment by protoism** [comment]
@keinhaar do you feel like contributing a PR for this? I'm afraid that it won't be that easy... 
@niloc132 you're the maven guru, here. Do we need to create a new repo, or it makes sense to have different maven targets?

**Comment by niloc132** [comment]
>  @niloc132 you're the maven guru, here. Do we need to create a new repo, or it makes sense to have different maven targets?

I'm afraid I don't understand what you're asking me - we already have a gwt.version property that can in theory be set to different versions of GWT to test with (but shouldn't change build output as far as I'm aware).

> The plugin still needs an SDK, which is more than the bunch of GWT jars already in the classpath,

What else is the SDK if not just the jars? Something that isn't shipped in GWT's jars (user, servlet, dev), but needed by the plugin itself? Or are you saying that the plugin can't function without the jars, and so the jars/SDK are needed for more than the project classpath?

**Comment by keinhaar** [comment]
@niloc132 @protoism 
I've managed to create a project, that builds only the SDK plugins and features.
I've uploaded the resulting update site here: http://www.keeitsi.com/software/eclipse-plugins/gwt-sdk-plugins
I added it to my composite update site (http://www.keeitsi.com/software/eclipse-plugins), and now i can install gwt-plugin, sdbg, and sdks, all at once.
Also added SDK 2.10.0.
Because the SDK Plugins do not depend on anything else, there is no need to have them rebuild if new gwt-eclipse-plugin versions are build.

So it seems to work.

Should we then create a new sub project? gwt-sdk-plugins?

**Comment by keinhaar** [comment]
@protoism, @niloc132 
I've created a separate sdk project here as a starting point:

https://github.com/keinhaar/gwt-sdk-plugins

It builds the 2.7 up to the 2.10 SDKs. You may review it there, but it should better be a separate project here: https://github.com/gwt-plugins.

Because of my limited maven knowledge, i've just stripped down the gwt-eclipse-plugin project until it only build the SDKs, and added 2.10 support.

I don't see any need for an aggregator repo, because the composite site consists of only 2 files with static references to the real repos.

> I wonder if integration tests require a plugin, but I doubt it.
I've removed the sdk builds from my gwt-eclipse-plugin branch, and it builds just fine. Tests are running.

**Comment by protoism** [comment]
Great job. @keinhaar

Il mer 12 apr 2023, 19:33 keinhaar ***@***.***> ha scritto:

**Comment by protoism** [comment]
I think it's time to revive this ticket.
@keinhaar, do you feel like contributing a PR?

**Comment by keinhaar** [comment]
We will need a new Project, but yes, i will look at my prototype and start right after 4.1 is out.

**Comment by keinhaar** [comment]
@protoism 
I started a new project "gwt-sdk-plugins" as part of gwt-plugins.
I've added my prototype from the last year, but updated it with removing SDK 2.7, and adding 2.11 as you did it here.
The update repository builds, and SDKs can be installed to eclipse (testet 2024-06).
Currently there is no code signing or automated build pipeline.
Please check if you can build too.
